### PR TITLE
gc_worker: fix bug of saving gc safe point to pd before do distributed gc

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
Fix https://github.com/pingcap/tidb/issues/10674 bug of saving gc safe point to pd before do distributed gc.

### Check List 

Tests 

 - No test. actually, I don't know how to add test for this...

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

### CODE REVIEW

**Comment by crazycs520** [comment]
@MyonKeminta @tiancaiamao @jackysp  PTAL

**Comment by MyonKeminta** [comment]
LGTM, and by the way I hope there will be some tests to cover it...
Sorry for not noticing this bug when I was working on distributed GC..

**Comment by winkyao** [comment]
Could we add some test cases？whatever ut or integration test?

File: store/tikv/gcworker/gc_worker.go [hunk 1]
```diff
@@ -1038,11 +1050,6 @@ func (w *GCWorker) genNextGCTask(bo *tikv.Backoffer, safePoint uint64, key kv.Ke
 func (w *GCWorker) doGC(ctx context.Context, safePoint uint64, concurrency int) error {
 	metrics.GCWorkerCounter.WithLabelValues("do_gc").Inc()
 
-	err := w.saveSafePoint(w.store.GetSafePointKV(), tikv.GcSavedSafePoint, safePoint)
```
Comment (tiancaiamao) [inline store/tikv/gcworker/gc_worker.go]: There are other functions that call `doGC`, see `RunGCJob`, if you simply remove `saveSafePoint` here, those function will be affected

Comment (winkyao) [reply store/tikv/gcworker/gc_worker.go]: but RunGCJob is useless now, @MyonKeminta PTAL

File: store/tikv/gcworker/gc_worker.go [hunk 2]
```diff
@@ -1038,11 +1050,6 @@ func (w *GCWorker) genNextGCTask(bo *tikv.Backoffer, safePoint uint64, key kv.Ke
...
-	if err != nil {
-		return errors.Trace(err)
-	}
-
 	// Sleep to wait for all other tidb instances update their safepoint cache.
 	time.Sleep(gcSafePointCacheInterval)
```
Comment (crazycs520) [inline store/tikv/gcworker/gc_worker.go]: @MyonKeminta Should I move this sleep after save safe point?

Comment (winkyao) [reply store/tikv/gcworker/gc_worker.go]: What if some tidb does not update their safepoint after we sleep(gcSafePointCacheInterval)?

File: store/tikv/gcworker/gc_worker.go (hunk 1 above)
Comment (tiancaiamao) [reply store/tikv/gcworker/gc_worker.go]: The API is not removed, you can't keep others from calling it. @winkyao

File: store/tikv/gcworker/gc_worker.go (hunk 2 above)
Comment (tiancaiamao) [reply store/tikv/gcworker/gc_worker.go]: Definitely no!

File: store/tikv/gcworker/gc_worker.go (hunk 1 above)
Comment (crazycs520) [reply store/tikv/gcworker/gc_worker.go]: already added save safe point time here.

File: store/tikv/gcworker/gc_worker.go (hunk 2 above)
Comment (disksing) [reply store/tikv/gcworker/gc_worker.go]: @winkyao Some tidb will report "ts may fall behink safepoint". It works kind like the DDL lease.

File: store/tikv/gcworker/gc_worker.go (hunk 1 above)
Comment (disksing) [reply store/tikv/gcworker/gc_worker.go]: @winkyao It is not useless. Comments state that _It is exported for kv api_.

**Comment by disksing** [comment]
LGTM.

File: store/tikv/gcworker/gc_worker.go (hunk 1 above)
Comment (MyonKeminta) [reply store/tikv/gcworker/gc_worker.go]: Yes, the RunGCJob and RunDistributedGCJob is useful for kv only clusters.

File: store/tikv/gcworker/gc_worker.go (hunk 2 above)
Comment (MyonKeminta) [reply store/tikv/gcworker/gc_worker.go]: @tiancaiamao could you please explain what the problem is?

File: store/tikv/gcworker/gc_worker.go [hunk 3]
```diff
@@ -1302,6 +1309,12 @@ func RunGCJob(ctx context.Context, s tikv.Storage, safePoint uint64, identifier
 	if concurrency <= 0 {
 		return errors.Errorf("[gc worker] gc concurrency should greater than 0, current concurrency: %v", concurrency)
 	}
+
+	err = gcWorker.saveSafePoint(gcWorker.store.GetSafePointKV(), tikv.GcSavedSafePoint, safePoint)
+	if err != nil {
```
Comment (MyonKeminta) [inline store/tikv/gcworker/gc_worker.go]: please also add this to RunDistributedGCJob.

Comment (crazycs520) [reply store/tikv/gcworker/gc_worker.go]: done.

**Comment by crazycs520** [comment]
Test failed in race test, It is no business with this PR, the race test failed will fix in PR https://github.com/pingcap/tidb/pull/10941

**Pull request merged**